example/mapboxExample.js

```diff
@@ -0,0 +1,176 @@
+import * as THREE from 'three';
+import CameraSync, {
+	projectToWorld,
+	projectedUnitsPerMeter,
+} from './mapboxExampleCamera.js';
+import { DebugTilesRenderer as TilesRenderer } from '../src/index.js';
+import { GUI } from 'three/examples/jsm/libs/lil-gui.module.min.js';
+
+const params = {
+
+	'accessToken': 'pk.eyJ1IjoieGlheGlhbmdmbmVnIiwiYSI6ImNscTRzc245NTA5c3cya3BhdXA4MDY0NWQifQ._9T_J9z7Hvcd00nQL3UuSw',
+	'reload': reload,
+
+};
+
+// GUI
+const gui = new GUI();
+gui.width = 300;
+
+const mapboxglOptions = gui.addFolder( 'mapboxgl' );
+mapboxglOptions.add( params, 'accessToken' );
+mapboxglOptions.add( params, 'reload' );
+mapboxglOptions.open();
+
+function rotationBetweenDirections( dir1, dir2 ) {
+
+	const rotation = new THREE.Quaternion();
+	const a = new THREE.Vector3().crossVectors( dir1, dir2 );
+	rotation.x = a.x;
+	rotation.y = a.y;
+	rotation.z = a.z;
+	rotation.w = 1 + dir1.clone().dot( dir2 );
+	rotation.normalize();
+
+	return rotation;
+
+}
+
+let map;
+const origin = [ 116, 35 ];
+
+init();
+
+function init() {
+
+	// TO MAKE THE MAP APPEAR YOU MUST
+	// ADD YOUR ACCESS TOKEN FROM
+	// https://account.mapbox.com
+	window.mapboxgl.accessToken = params.accessToken;
+	map = new window.mapboxgl.Map( {
+		container: 'map',
+		style: 'mapbox://styles/mapbox/dark-v9',
+		zoom: 17.86614600777933,
+		pitch: 70,
+		bearing: - 40,
+		center: origin,
+	} );
+
+	map.on( 'load', () => {
+
+		let renderer, scene, camera, world, tiles, tilesGroup;
+
+		map.addLayer( {
+			id: 'custom_layer',
+			type: 'custom',
+			onAdd: function ( map, mbxContext ) {
+
+				renderer = new THREE.WebGLRenderer( {
+					alpha: true,
+					antialias: true,
+					canvas: map.getCanvas(),
+					context: mbxContext,
+				} );
+
+				renderer.shadowMap.enabled = true;
+				renderer.autoClear = false;
+
+				scene = new THREE.Scene();
+				camera = new THREE.Camera();
+
+				world = new THREE.Group();
+				scene.add( world );
+
+				// lights
+				const dirLight = new THREE.DirectionalLight( 0xffffff, 0.8 );
+				dirLight.position.set( 1, 2, 3 );
+				world.add( dirLight );
+
+				const ambLight = new THREE.AmbientLight( 0xffffff, 1 );
+				world.add( ambLight );
```

Comment:
The tileset used is using basic materials which don't use lighting - I think we can remove these for now.

Reply:
ok